File: 11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        long result = 0;
        int left = 0;
        int right = height.size() - 1;
         
        while (left < right) {
            long water = min(height[left], height[right]) * (right - left);
            result = max(result, water);
            height[left] < height[right] ? left++ : right--;
        }
        return result;
    }
};

File: 11.rb
# @param {Integer[]} height
# @return {Integer}
def max_area(height)
  result = 0
  left = 0
  right = height.size - 1

  while left != right
    water = [height[left], height[right]].min * (right - left)
    result = [result, water].max
    height[left] < height[right] ? (left += 1) : (right -= 1)
  end
  result
end
